Generated constructors incorrectly accept positional arguments

The annotated `__init__` method generated for a protobuf message accepts positional arguments, but the underlying implementation accepts only named arguments. The first parameter after `self` should be a nameless `*` parameter. This will tell mypy and other type checkers that positional arguments are not allowed when calling the constructor.

For example, a protobuf message with a `string: name` field will generate the following annotated stub:
```
class MyMessage(google___protobuf___message___Message):
    name = ... # type: typing___Text

    def __init__(self,
        name : typing___Optional[typing___Text] = None) -> None: ...
```

If callers attempt to construct this message by calling `MyMessage('bob')`, a runtime error occurs indicating that positional arguments are not allowed for this call.

To match the underlying implementation, it should be:
```
    def __init__(self, *,
        name : typing___Optional[typing___Text] = None) -> None: ...
```

I was going to write a patch as suggested. Before changing any code, I ran the test script, and I'm seeing failures. Are the tests supposed to be in a passing state, or is this a known issue? Maybe it's a problem with my configuration?

Here's a snippet of the output I see when I run `run_test.sh`. I'm using MacOS and Python 3.6.
```
        with pytest.raises(ValueError, match="Unknown field garbage."):
>           s_untyped.HasField("garbage")
E           ValueError: Protocol message Simple1 has no field garbage.

test/test_generated_mypy.py:117: ValueError

...
        # Erase the types to verify that incorrect inputs fail at runtime
        # Each test here should be duplicated in test_negative to ensure mypy fails it too
        s_untyped = s  # type: Any
        with pytest.raises(ValueError, match="Unknown field garbage."):
>           s_untyped.HasField(u"garbage")
E           ValueError: Protocol message SimpleProto3 has no field garbage.

test/test_generated_mypy.py:135: ValueError
```

--- a/python/protoc-gen-mypy
+++ b/python/protoc-gen-mypy
@@ -179,6 +179,10 @@ class PkgWriter(object):
                 # Constructor
                 l("def __init__(self,")
                 with self._indent():
+                    if len(desc.field) > 0:
+                        # Only positional args allowed
+                        # See https://github.com/dropbox/mypy-protobuf/issues/71
+                        l("*,")
                     # Required args
                     for field in [f for f in desc.field if f.label == d.FieldDescriptorProto.LABEL_REQUIRED]:
                         l("{} : {},", field.name, self.python_type(field))

--- a/test/proto/Capitalized/Capitalized_pb2.pyi.expected
+++ b/test/proto/Capitalized/Capitalized_pb2.pyi.expected
@@ -17,6 +17,7 @@ class lower(google___protobuf___message___Message):
     a = ... # type: int
 
     def __init__(self,
+        *,
         a : typing___Optional[int] = None,
         ) -> None: ...
     @classmethod
@@ -34,6 +35,7 @@ class Upper(google___protobuf___message___Message):
     def Lower(self) -> lower: ...
 
     def __init__(self,
+        *,
         Lower : typing___Optional[lower] = None,
         ) -> None: ...
     @classmethod
@@ -53,6 +55,7 @@ class lower2(google___protobuf___message___Message):
     def upper(self) -> Upper: ...
 
     def __init__(self,
+        *,
         upper : typing___Optional[Upper] = None,
         ) -> None: ...
     @classmethod

--- a/test/proto/inner/inner_pb2.pyi.expected
+++ b/test/proto/inner/inner_pb2.pyi.expected
@@ -21,6 +21,7 @@ class Inner(google___protobuf___message___Message):
     a = ... # type: test___proto___test3_pb2___OuterEnum
 
     def __init__(self,
+        *,
         a : typing___Optional[test___proto___test3_pb2___OuterEnum] = None,
         ) -> None: ...
     @classmethod

--- a/test/proto/nested/nested_pb2.pyi.expected
+++ b/test/proto/nested/nested_pb2.pyi.expected
@@ -29,6 +29,7 @@ class Nested(google___protobuf___message___Message):
     a = ... # type: test___proto___test3_pb2___OuterEnum
 
     def __init__(self,
+        *,
         a : typing___Optional[test___proto___test3_pb2___OuterEnum] = None,
         ) -> None: ...
     @classmethod
@@ -80,6 +81,7 @@ class AnotherNested(google___protobuf___message___Message):
         ne2 = ... # type: AnotherNested.NestedMessage.NestedEnum2
 
         def __init__(self,
+            *,
             s : typing___Optional[typing___Text] = None,
             b : typing___Optional[bool] = None,
             ne : typing___Optional[AnotherNested.NestedEnum] = None,

--- a/test/proto/nopackage_pb2.pyi.expected
+++ b/test/proto/nopackage_pb2.pyi.expected
@@ -36,6 +36,7 @@ class NoPackage2(google___protobuf___message___Message):
     def np_rep(self) -> google___protobuf___internal___containers___RepeatedCompositeFieldContainer[NoPackage]: ...
 
     def __init__(self,
+        *,
         np : typing___Optional[NoPackage] = None,
         np_rep : typing___Optional[typing___Iterable[NoPackage]] = None,
         ) -> None: ...

--- a/test/proto/test3_pb2.pyi.expected
+++ b/test/proto/test3_pb2.pyi.expected
@@ -47,6 +47,7 @@ class OuterMessage3(google___protobuf___message___Message):
     a_bool = ... # type: bool
 
     def __init__(self,
+        *,
         a_bool : typing___Optional[bool] = None,
         ) -> None: ...
     @classmethod
@@ -71,6 +72,7 @@ class SimpleProto3(google___protobuf___message___Message):
     def outer_message(self) -> OuterMessage3: ...
 
     def __init__(self,
+        *,
         a_string : typing___Optional[typing___Text] = None,
         a_repeated_string : typing___Optional[typing___Iterable[typing___Text]] = None,
         outer_enum : typing___Optional[OuterEnum] = None,

--- a/test/proto/test_no_generic_services_pb2.pyi.expected
+++ b/test/proto/test_no_generic_services_pb2.pyi.expected
@@ -17,6 +17,7 @@ class Simple3(google___protobuf___message___Message):
     a_string = ... # type: typing___Text
 
     def __init__(self,
+        *,
         a_string : typing___Text,
         ) -> None: ...
     @classmethod

--- a/test/proto/test_pb2.pyi.expected
+++ b/test/proto/test_pb2.pyi.expected
@@ -131,6 +131,7 @@ class Simple1(google___protobuf___message___Message):
     def nested_message(self) -> test___proto___nested___nested_pb2___AnotherNested.NestedMessage: ...
 
     def __init__(self,
+        *,
         a_string : typing___Optional[typing___Text] = None,
         a_repeated_string : typing___Optional[typing___Iterable[typing___Text]] = None,
         a_boolean : typing___Optional[bool] = None,
@@ -165,6 +166,7 @@ class Simple2(google___protobuf___message___Message):
     a_string = ... # type: typing___Text
 
     def __init__(self,
+        *,
         a_string : typing___Text,
         ) -> None: ...
     @classmethod

